posthog/temporal/data_imports/pipelines/mongo/mongo.py

```diff
@@ -14,61 +13,133 @@
-                private_key=data.get("ssh_tunnel_auth_type_private_key"),
-            )
-        # Handle SSH tunnel configuration from nested format
-        elif "ssh_tunnel" in data and isinstance(data["ssh_tunnel"], dict):
-            ssh_tunnel_data = data["ssh_tunnel"]
-            if ssh_tunnel_data.get("enabled"):
-                auth_data = ssh_tunnel_data.get("auth", {})
-                ssh_tunnel_config = SSHTunnelConfig(
-                    enabled=ssh_tunnel_data.get("enabled", False),
-                    host=ssh_tunnel_data.get("host"),
-                    port=ssh_tunnel_data.get("port"),
-                    auth_type=auth_data.get("type"),
-                    username=auth_data.get("username"),
-                    password=auth_data.get("password"),
-                    passphrase=auth_data.get("passphrase"),
-                    private_key=auth_data.get("private_key"),
-                )
-
-        return cls(
-            connection_string=data["connection_string"],
-            ssh_tunnel=ssh_tunnel_config,
+
+
+def _process_nested_value(value: Any) -> Any:
+    """Process a nested value, converting ObjectIds to strings."""
+    if isinstance(value, ObjectId):
+        return str(value)
+    elif isinstance(value, dict):
+        return _process_nested_object(value)
+    elif isinstance(value, list):
+        return [_process_nested_value(item) for item in value]
+    else:
+        return value
+
+
+def get_indexes(connection_string: str, collection_name: str) -> list[str]:
+    """Get all indexes for a MongoDB collection."""
+    try:
+        connection_params = _parse_connection_string(connection_string)
+        client = _create_mongo_client(connection_string, connection_params)
+        db = client[connection_params["database"]]
+        collection = db[collection_name]
+
+        index_cursor = collection.list_indexes()
+        return [field for index in index_cursor for field in index["key"].keys()]
+    except Exception:
+        return []
+
+
+def _process_nested_object(obj: dict) -> dict:
+    """Process a nested object, converting ObjectIds to strings recursively."""
+    processed = {}
+    for key, value in obj.items():
+        processed[key] = _process_nested_value(value)
+    return processed
+
+
+def _build_query(
+    is_incremental: bool,
+    incremental_field: Optional[str],
+    incremental_field_type: Optional[IncrementalFieldType],
+    db_incremental_field_last_value: Optional[Any],
+) -> dict[str, Any]:
+    query = {}
+
+    if not is_incremental:
+        return query
+
+    if incremental_field is None or incremental_field_type is None:
+        raise ValueError("incremental_field and incremental_field_type can't be None")
+
+    if db_incremental_field_last_value is None:
+        db_incremental_field_last_value = incremental_type_to_initial_value(incremental_field_type)
+
+    query = {incremental_field: {"$gte": db_incremental_field_last_value, "$exists": True}}
```

Comment:
Do you not need to wrap the incremental last value in a `ObjectId(...)`? Mongo typically doesn't like you comparing objectIds to plain strings